src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,205 @@
-""" This is a dummy example """
+import requests
+import pandas as pd
+import matplotlib.pyplot as plt
+from itertools import cycle
+
+# Base URL for accessing the climate data API
+API_URL = "https://climate-api.open-meteo.com/v1/climate?"
+
+# Dictionary to store geographic coordinates for specific cities
+COORDINATES = {
+    "Madrid": {"latitude": 40.416775, "longitude": -3.703790},
+    "London": {"latitude": 51.507351, "longitude": -0.127758},
+    "Rio": {"latitude": -22.906847, "longitude": -43.172896},
+}
+
+# Define the climate metrics to fetch
+VARIABLES = "temperature_2m_mean,precipitation_sum,soil_moisture_0_to_10cm_mean"  # noqa
+MODELS = "CMCC_CM2_VHR4,FGOALS_f3_H,HiRAM_SIT_HR,MRI_AGCM3_2_S,EC_Earth3P_HR,MPI_ESM1_2_XR,NICAM16_8S"  # noqa
+VARIABLES_MODELS = [
+    f"{metric}_{model}"
+    for metric in VARIABLES.split(",")
+    for model in MODELS.split(",")
+]
+
+
+def get_data_meteo_api(city, start_year="1950-01-01", end_year="2050-12-31"):
```

Comment:
Typing is extremely important. You should type both inputs and outputs of functions. The small comment explaining the function is a good practice.